vyper/cli/vyper_compile.py

```diff
@@ -135,16 +138,33 @@ def _parse_args(argv):
         args.ovm,
     )
 
-    if output_formats == ("combined_json",):
-        print(json.dumps(compiled))
-        return
 
-    for contract_data in compiled.values():
-        for data in contract_data.values():
-            if isinstance(data, (list, dict)):
-                print(json.dumps(data))
-            else:
-                print(data)
+    if not args.output_path:
+        if output_formats == ("combined_json",):
+            print(json.dumps(compiled))
+            return
+
+        for contract_data in compiled.values():
+            for data in contract_data.values():
+                if isinstance(data, (list, dict)):
+                    print(json.dumps(data))
+
+    else:
+        if output_formats == ("combined_json",):
+            with open(args.output_path,"w") as f:
+                f.write(json.dumps(compiled))
+                return
+                
+        for contract_data in compiled.values():
+            for data in contract_data.values():
+                if isinstance(data, (list, dict)):                    
+                    with open(args.output_path,"w") as f:
+                        f.write(json.dumps(data))
+                elif "bytecode" in contract_data.keys():
```

Comment:
What is the purpose of this branch?

Reply:
this is for writing ABI files. would it be better to catch each case?

Reply:
Maybe I'm being dense here but what combination of flags activates this branch?

Reply:
thanks for the review, will make the suggest change so it's easier to read. I think in that case is pretty small just direct output to file or stdout and leave previous behavior.